src/glade.py

```diff
@@ -49,16 +59,25 @@ def __str__(self):
             assert False
 
 class Alt(Regex):
-    def __init__(self, a1, a2): self.a1, self.a2 = a1, a2
+    def __init__(self, a1, a2, extra): 
+        self.a1 = a1 
+        self.a2 = a2
+        self.extra = extra  # extra data used to mark if this object needs to be considered in the next check (if equals to 9) or not (equqls to 0). 
+                            # That is, whether it's a part of the Context or not. See section 4.3
```

Comment:
Would `is_part_of_context` a better name for `extra`?

Reply:
Also, simply copy and paste the relevant text from paper (indicate it is Section 4.3, and paste the text that is relevant here.)